pkg/config/config.go

```diff
@@ -0,0 +1,123 @@
+package config
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path/filepath"
+)
+
+const (
+	ConfigFileName       = "weave-gitops-config.json"
+	WrongConfigFormatMsg = `Your CLI configuration for Weave GitOps should represent a JSON object format:
+	{
+		"analytics": true,
+		"userId": ""
+	}
+Please set you configuration with: gitops set config`
+)
+
+type GitOpsCLIConfig struct {
+	Analytics bool   `json:"analytics"`
+	UserID    string `json:"userId"`
+}
+
+// GetConfigPath composes the path of the config file for the GitOps CLI
+func GetConfigPath(filename string) (string, error) {
+	var err error
+
+	userConfigDir, err := os.UserConfigDir()
+	if err != nil {
+		return "", fmt.Errorf("user config directory could not be determined: %w", err)
+	}
+
+	configPath := filepath.Join(userConfigDir, filename)
+
+	return configPath, nil
+}
+
+// OpenConfigFile opens existing config file or creates one if it does not exist
+func OpenConfigFile(configPath string, shouldCreate bool) (*os.File, error) {
+	_, err := os.Stat(configPath)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) || !shouldCreate {
+			return nil, err
+		}
+	}
+
+	var configFile *os.File
+
+	if shouldCreate {
+		configFile, err = os.Create(configPath)
+		if err != nil {
+			return nil, fmt.Errorf("error creating config file: %w", err)
+		}
+	} else {
+		configFile, err = os.Open(configPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open config file: %w", err)
+		}
+	}
+
+	return configFile, nil
+}
+
+// ReadConfigData reads the data from the config file
+func ReadConfigData(configFile *os.File) ([]byte, error) {
+	data, err := io.ReadAll(configFile)
+	if err != nil {
+		return nil, err
+	}
+
+	return data, nil
+}
+
+// ParseConfig unmarshals the CLI configuration for Weave GitOps from the JSON blob
+func ParseConfig(data []byte, config *GitOpsCLIConfig) error {
+	err := json.Unmarshal(data, &config)
+	if err != nil {
+		return fmt.Errorf("error parsing config JSON: %w", err)
+	}
+
+	return nil
+}
+
+// WriteConfig reads the CLI configuration for Weave GitOps from the config file
+func WriteConfig(config *GitOpsCLIConfig, configPath string) error {
+	data, err := json.MarshalIndent(&config, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error encoding config: %w", err)
+	}
+
+	configFile, err := OpenConfigFile(configPath, true)
+	if err != nil {
+		return fmt.Errorf("error opening config file: %w", err)
+	}
+
+	defer configFile.Close()
+
+	_, err = configFile.Write(data)
+	if err != nil {
+		return fmt.Errorf("error writing to config file: %w", err)
+	}
+
+	return nil
+}
+
+var letters = []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+// GenerateUserID generates a string of specified length made of random characters and encodes it in base64 format
+func GenerateUserID(numChars int, seed int64) string {
```

Comment:
Given that you've chosen to make the original ID printable, there's no real need to base64 encode it again - you could add `0123456789+/` to the letters array, and it would already be a base64 representation of a random value.

Reply:
Added the characters and removed the additional base24 encoding.

> Given that you've chosen to make the original ID printable

What would making the original ID non-printable look like? Some binary operations on numbers or what?

Reply:
```
b := make([]byte, 10)
_, err := rand.Read(b)
```

Reply:
Cool! Would you prefer to use it instead?